cpp/benchmarks/copying/scatter.cu

```diff
@@ -14,78 +14,49 @@
 };
 
 template <class TypeParam, bool coalesce>
 void BM_scatter(benchmark::State& state)
 {
-  const cudf::size_type source_size{(cudf::size_type)state.range(0)};
-  const auto n_cols = (cudf::size_type)state.range(1);
-
-  // Every element is valid
-  auto data = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return i; });
+  auto const source_size{static_cast<cudf::size_type>(state.range(0))};
+  auto const n_cols{static_cast<cudf::size_type>(state.range(1))};
 
   // Gather indices
-  std::vector<cudf::size_type> host_map_data(source_size);
-  std::iota(host_map_data.begin(), host_map_data.end(), 0);
+  auto scatter_map_table =
+    create_sequence_table({cudf::type_to_id<cudf::size_type>()}, 1, row_count{source_size});
+  auto scatter_map = scatter_map_table->get_column(0).mutable_view();
 
   if (coalesce) {
-    std::reverse(host_map_data.begin(), host_map_data.end());
+    thrust::reverse(
+      thrust::device, scatter_map.begin<cudf::size_type>(), scatter_map.end<cudf::size_type>());
   } else {
-    std::random_shuffle(host_map_data.begin(), host_map_data.end());
+    thrust::shuffle(thrust::device,
+                    scatter_map.begin<cudf::size_type>(),
+                    scatter_map.end<cudf::size_type>(),
+                    thrust::default_random_engine());
   }
 
-  cudf::test::fixed_width_column_wrapper<cudf::size_type> scatter_map(host_map_data.begin(),
-                                                                      host_map_data.end());
-
-  std::vector<cudf::test::fixed_width_column_wrapper<TypeParam>> source_column_wrappers;
-  std::vector<cudf::column_view> source_columns(n_cols);
-
-  std::vector<cudf::test::fixed_width_column_wrapper<TypeParam>> target_column_wrappers;
-  std::vector<cudf::column_view> target_columns(n_cols);
-
-  std::generate_n(std::back_inserter(source_column_wrappers), n_cols, [=]() {
-    return cudf::test::fixed_width_column_wrapper<TypeParam>(data, data + source_size);
-  });
-  std::transform(source_column_wrappers.begin(),
-                 source_column_wrappers.end(),
-                 source_columns.begin(),
-                 [](auto const& col) { return static_cast<cudf::column_view>(col); });
-
-  std::generate_n(std::back_inserter(target_column_wrappers), n_cols, [=]() {
-    return cudf::test::fixed_width_column_wrapper<TypeParam>(data, data + source_size);
-  });
-  std::transform(target_column_wrappers.begin(),
-                 target_column_wrappers.end(),
-                 target_columns.begin(),
-                 [](auto const& col) { return static_cast<cudf::column_view>(col); });
-
-  cudf::table_view source_table{source_columns};
-  cudf::table_view target_table{target_columns};
+  // Every element is valid
+  auto source_table =
+    create_sequence_table({cudf::type_to_id<TypeParam>()}, n_cols, row_count{source_size});
+  auto target_table =
+    create_sequence_table({cudf::type_to_id<TypeParam>()}, n_cols, row_count{source_size});
```

Comment:
This snippet indicates to me that we should factor out the dtype cycling from `create_sequence_table`. We only need to call that once (rather than doing it in each call to `create_sequence_table`), and doing so explicitly before calling `create_sequence_table` would be clearer in my opinion.

Reply:
updated.